src/varcache.c

```diff
@@ -99,12 +180,17 @@ static int apply_var(PktBuf *pkt, const char *key,
 	if (strcasecmp(cval->str, sval->str) == 0)
 		return 0;
 
-	/* the string may have been taken from startup pkt */
-	if (!pg_quote_literal(qbuf, cval->str, sizeof(qbuf)))
+	if (strcasecmp("search_path", key) == 0) {
```

Comment:
Needs a comment why search_path is special.

Reply:
Actually it turns out that this needs even more special quoting...

The correct way of quoting a GUC which has the GUC_LIST_QUOTE flag is by doing what pgdump does:
https://github.com/postgres/postgres/blob/555b929bbe7bd2f228e804d2c5bf4a246a319358/src/bin/pg_dump/dumputils.c#L847-L880

Otherwise things break with long search paths or an empty list:
```
> set search_path = aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa;
NOTICE:  42622: identifier "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" will be truncated to "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```
and 
```
> set search_path = "";
ERROR:  42601: zero-length delimited identifier at or near """"
LINE 1: set search_path = "";
```

Probably best to copy this quoting logic from pgdump to a separate file.

Reply:
As discussed, let's only consider the empty string case to keep the code simple. Long identifiers being truncated is not important, schema identifiers are already bounded to the truncated length anyway.